src/compiler/scala/tools/nsc/backend/jvm/BTypes.scala

```diff
@@ -795,6 +776,9 @@ abstract class BTypes {
       } while (fcs == null)
       fcs
     }
+
+    override val toASMType: asm.Type = super.toASMType
+    override val toString: String = super.toString
```

Comment:
not sure if we get enough reuse at this level to warrant he allocation. If we drop to the implementations. For ClassBType it seems a likely win (albeit inefficient). 
For others, as they are not cached I dont see that we will get any reuse, so it not worth eagerly calculating.

Reply:
its reasonably easy to cache ArrayBTypes. Maybe thats worth doing
```
object ArrayBType {
    private val cache = new ju.concurrent.ConcurrentHashMap[BType, ArrayBType]
    
    def apply(componentType: BType): ArrayBType =
      cache.computeIfAbsent(componentType, componentType => new ArrayBType(componentType))
  }
```
seems that LMF and others uses ArrayBTypes so it probably is quite common

Reply:
This only applied to `ClassBType`. I see your point that the eager caching could end up creating values for `ClassBType.toString` that wouldn't otherwise been needed. I've pushed a commit to make this lazy.